staging/src/k8s.io/apiserver/pkg/util/webhook/client.go

```diff
@@ -179,6 +180,11 @@ func (cm *ClientManager) HookClient(cc ClientConfig) (*rest.RESTClient, error) {
 		if len(cfg.TLSClientConfig.ServerName) == 0 {
 			cfg.TLSClientConfig.ServerName = serverName
 		}
+		// request to Services should not be proxied #107446
+		// https://github.com/golang/net/blob/c6ed85c7a12db1bd15e993fd3ae4700b2e9f2c84/http/httpproxy/proxy.go#L112-L114
+		// A nil URL and nil error are returned if no proxy is defined in the environment,
+		// or a proxy should not be used for the given request, as defined by NO_PROXY
+		cfg.Proxy = func(req *http.Request) (*url.URL, error) { return nil, nil }
```

Comment:
I'm curious why users can't set NO_PROXY to include service domains as opposed to dropping proxy for all services here. In this way you can still enforce proxy for services if you need to, which may be relevant for control plane nodes in isolated/managed networks

Reply:
yeah, I was squinting at this trying to figure out if there were working clusters happily routing requests to services through a proxy that this would break... I don't know how to determine that

Reply:
this build always requests to `svcnname.svcnamespace.svc`, appending `svc` as the top level domain, adding it to NO_PROXY variable should work too, but is not easy to know that

Reply:
> I was squinting at this trying to figure out if there were working clusters happily routing requests to services through a proxy that this would break

the apiserver should be able to reach the services directly or via the egress routing (konnectivity), right? 
but now that you mention it somebody reached out a few weeks ago about something related, let me ask